### PR TITLE
Add access function for restoration state of alerting rule

### PR DESCRIPTION
**Pull request by qinxx108**
Was working on https://github.com/cortexproject/cortex/issues/4435
The issue that we have to enable the ruler HA on cortex is the activeAt attribute for alerting rule which the value need to be aligned between replicas. We was thinking about doing a continuously pulling for ALERTS_FOR_STATE metric and update local activeAt. By doing that we need the access of alerting rules' restoration state outside of prometheus package
Add access function for restoration state of alerting rule

### CODE REVIEW

**Comment by alvinlin123** [comment]
@qinxx108

**Comment by qinxx108** [comment]
@roidelapluie

File: rules/alerting.go [hunk 1]
```diff
@@ -291,6 +291,13 @@ func (r *AlertingRule) SetRestored(restored bool) {
 	r.restored = restored
 }
 
+// Restored returns the if restoration state of the alerting rule.
+func (r *AlertingRule) Restored() bool {
```
Comment (LeviHarrison) [inline rules/alerting.go:295]: `GetRestored` aligns with other functions' naming.

```suggestion
// GetRestored returns the restoration state of the alerting rule.
func (r *AlertingRule) GetRestored() bool {
```

Comment (qinxx108) [reply rules/alerting.go:295]: https://github.com/prometheus/prometheus/blob/main/rules/alerting.go#L178-L203 i was follow convention here. But it's your call :)

Comment (qinxx108) [reply rules/alerting.go:295]: I was following convention at here: https://github.com/prometheus/prometheus/blob/main/rules/alerting.go#L178-L203. But it's your call :)

Comment (LeviHarrison) [reply rules/alerting.go:295]: The functions directly above it use "Get", which is what I had looked at: https://github.com/prometheus/prometheus/blob/d80e2e83c252272ee586a47edb906179caff3bfc/rules/alerting.go#L262-L287

But after referring to [Effective Go](https://golang.org/doc/effective_go#Getters), it seems that the proper way is to not use "Get", so I think this is good. Thanks!

File: rules/alerting.go [hunk 2]
```diff
@@ -291,6 +291,13 @@ func (r *AlertingRule) SetRestored(restored bool) {
 	r.restored = restored
 }
 
+// Restored returns the if restoration state of the alerting rule.
```
Comment (LeviHarrison) [inline rules/alerting.go:294]:
```suggestion
// Restored returns the restoration state of the alerting rule.
```

**Comment by LeviHarrison** [comment]
Thanks!

**Pull request merged**